src/plugins/controls/public/control_group/control_group_renderer.tsx

```diff
@@ -8,41 +8,87 @@
 
 import uuid from 'uuid';
 import useLifecycles from 'react-use/lib/useLifecycles';
-import React, { useEffect, useMemo, useRef, useState } from 'react';
+import React, { useMemo, useRef, useState } from 'react';
 
 import { IEmbeddable } from '@kbn/embeddable-plugin/public';
+import { useReduxContainerContext } from '@kbn/presentation-util-plugin/public';
 
 import { pluginServices } from '../services';
-import { getDefaultControlGroupInput } from '../../common';
-import { ControlGroupInput, ControlGroupOutput, CONTROL_GROUP_TYPE } from './types';
+import { ControlPanelState, getDefaultControlGroupInput } from '../../common';
+import {
+  ControlGroupInput,
+  ControlGroupOutput,
+  ControlGroupReduxState,
+  CONTROL_GROUP_TYPE,
+} from './types';
 import { ControlGroupContainer } from './embeddable/control_group_container';
+import { DataControlInput } from '../types';
+import { getCompatibleControlType, getNextPanelOrder } from './embeddable/control_group_helpers';
+import { controlGroupReducers } from './state/control_group_reducers';
+
+const ControlGroupInputBuilder = {
+  addDataControlFromField: async (
+    initialInput: Partial<ControlGroupInput>,
+    newPanelInput: {
```

Comment:
@nreese, I'm appreciating your suggestion to expose API methods here for individual control types, but I think we should also keep this generic one for simplicity.

In the individual control type builders, we would be able to take `selections`, in case a consumer wanted to build an Options List control for instance with some options pre-selected.